src/backend/distributed/deparser/qualify_view_stmt.c

```diff
@@ -38,13 +38,18 @@ QualifyDropViewStmt(Node *node)
 
 		if (schemaName == NULL)
 		{
-			char *objname = NULL;
-			Oid schemaOid = QualifiedNameGetCreationNamespace(possiblyQualifiedViewName,
-															  &objname);
-			schemaName = get_namespace_name(schemaOid);
-			List *qualifiedViewName = list_make2(makeString(schemaName),
-												 makeString(viewName));
-			qualifiedViewNames = lappend(qualifiedViewNames, qualifiedViewName);
+			RangeVar *viewRangeVar = makeRangeVarFromNameList(possiblyQualifiedViewName);
+			Oid viewOid = RangeVarGetRelid(viewRangeVar, AccessExclusiveLock, true);
+
+			if (OidIsValid(viewOid))
+			{
+				Oid schemaOid = get_rel_namespace(viewOid);
+				schemaName = get_namespace_name(schemaOid);
+
+				List *qualifiedViewName = list_make2(makeString(schemaName),
+													 makeString(viewName));
+				qualifiedViewNames = lappend(qualifiedViewNames, qualifiedViewName);
+			}
```

Comment:
There is a bit of a weird case here now, where the name silently drops from the drop statement if the view can't be resolved.

I would expect the original name to be retained on the list. Although this might cause unintended behaviour if that name would be send to the workers as is. (it might exist in the default search_path, but not in the search_path we are currently in. Meaning when we send it to the workers it could actually remove an object)

Could we at least document this here in a comment, and or `elog(ERROR, ... )` if we expect this to never happen, based on other interactions of the code where we call this?

From a correctness perspective of this function I would expect the function to keep unresolved objects in this list with the name as provided by the user. Relying on the rest of Citus to take the name out (given a non-existing object it is not distributed by definition) before deserializing and sending the commands to the workers.

Reply:
I've updated the code to keep nonexistent views in the list without trying to qualify them.